File: NewYearChaos/NewYearChaos.java
import java.io.*;
import java.math.*;
import java.security.*;
import java.text.*;
import java.util.*;
import java.util.concurrent.*;
import java.util.function.*;
import java.util.regex.*;
import java.util.stream.*;
import static java.util.stream.Collectors.joining;
import static java.util.stream.Collectors.toList;

class Result {

    /*
     * Complete the 'minimumBribes' function below.
     *
     * The function accepts INTEGER_ARRAY q as parameter.
     */

    public static void minimumBribes(List<Integer> q) {
    // Write your code here
        int bribe = 0;
        
        for(int i = 0; i<q.size(); i++){
            int person = q.get(i);
            //if current person and supposed position + 1 is more than 2
            if(person - (i + 1) > 2){ //when chaotic
                System.out.println("Too chaotic");
                return;
            }
            for(int j = Math.max(0, person - 2); j<i; j++){ //0 because for 1st index; loops two times to check if this person behind is before the person
                //check if current person is from the later queue
                if(q.get(j) > person){
                    bribe++;
                }
            }
        }
        System.out.println(bribe);
    }

}

public class Solution {
    public static void main(String[] args) throws IOException {
        BufferedReader bufferedReader = new BufferedReader(new InputStreamReader(System.in));

        int t = Integer.parseInt(bufferedReader.readLine().trim());

        IntStream.range(0, t).forEach(tItr -> {
            try {
                int n = Integer.parseInt(bufferedReader.readLine().trim());

                List<Integer> q = Stream.of(bufferedReader.readLine().replaceAll("\\s+$", "").split(" "))
                    .map(Integer::parseInt)
                    .collect(toList());

                Result.minimumBribes(q);
            } catch (IOException ex) {
                throw new RuntimeException(ex);
            }
        });

        bufferedReader.close();
    }
}

File: NewYearChaos/NewYearChaos.js
'use strict';

process.stdin.resume();
process.stdin.setEncoding('utf-8');

let inputString = '';
let currentLine = 0;

process.stdin.on('data', function(inputStdin) {
    inputString += inputStdin;
});

process.stdin.on('end', function() {
    inputString = inputString.split('\n');

    main();
});

function readLine() {
    return inputString[currentLine++];
}

/*
 * Complete the 'minimumBribes' function below.
 *
 * The function accepts INTEGER_ARRAY q as parameter.
 */

function minimumBribes(q) {
    // Write your code here
    let bribe = 0;
    for(let i = 0; i<q.length; i++){
        let person = q[i];
        if (person - (i+1) > 2){ //chaotic
            console.log("Too chaotic");
            return;
        }
        for(let j = Math.max(0,person-2); j<i; j++){
            if(q[j]>person){
                bribe++;
            }
        }
    }
    console.log(bribe);
}

function main() {
    const t = parseInt(readLine().trim(), 10);

    for (let tItr = 0; tItr < t; tItr++) {
        const n = parseInt(readLine().trim(), 10);

        const q = readLine().replace(/\s+$/g, '').split(' ').map(qTemp => parseInt(qTemp, 10));

        minimumBribes(q);
    }
}
